partner_event/tests/test_event_registration.py

```diff
@@ -61,15 +62,24 @@ def test_create(self):
 
     def test_count_registrations(self):
         event_1 = self.event_0.copy()
+        self.registration_01.state = "draft"
+        self.registration_02.state = "draft"
+        self.partner_01.invalidate_recordset()
         self.assertEqual(self.partner_01.registration_count, 0)
         self.registration_01.state = "open"
         self.partner_01.invalidate_recordset()
         self.assertEqual(self.partner_01.registration_count, 1)
-        self.registration_02.state = "done"
+        self.registration_02.state = "open"
         self.registration_02.attendee_partner_id = self.partner_01
         self.registration_02.event_id = event_1
         self.partner_01.invalidate_recordset()
         self.assertEqual(self.partner_01.registration_count, 2)
+        self.registration_01.state = "cancel"
+        self.partner_01.invalidate_recordset()
+        self.assertEqual(self.partner_01.registration_count, 1)
+        self.registration_02.state = "done"
+        self.partner_01.invalidate_recordset()
+        self.assertEqual(self.partner_01.registration_count, 1)
```

Comment:
Didn't the test work as it was?

Reply:
No, the test as it was before threw the following error:

File "/opt/odoo/auto/addons/partner_event/tests/test_event_registration.py", line 64, in test_count_registrations
    self.assertEqual(self.partner_01.registration_count, 0)
AssertionError: 1 != 0

Reply:
Yes, the test is there to check that the module behaves in a certain way. If the tests don't work it could either be that the logic of the module has changed or that something in the core has changed and the tests need to be adapted. The latter is a stranger case and requires more argumentation :wink: